pisek/task_jobs/builder/strategies.py

```diff
@@ -249,10 +250,31 @@ def applicable_on_files(cls, build: "BuildConfig", sources: list[str]) -> bool:
 
     def _build(self) -> str:
         pas_flags = ["-gl", "-O3", "-Sg", "-o" + self.target, "-FE."]
-        return self._run_compilation(
+        self._run_subprocess(
             ["fpc"] + pas_flags + self.sources + self._build_section.comp_args,
             self._build_section.program_name,
         )
+        return self.target
+
+
+class Make(BuildStrategy):
+    name = BuildStrategyName.make
+
+    @classmethod
+    def applicable_on_files(cls, build: "BuildConfig", sources: list[str]) -> bool:
+        return False
+
+    @classmethod
+    def applicable_on_directory(cls, build: "BuildConfig", directory: str) -> bool:
+        return os.path.exists(os.path.join(directory, "Makefile"))
+
+    def _build(self) -> str:
+        directory = os.listdir()[0]
+        with ChangedCWD(directory):
```

Comment:
It might make sense to delete the target directory (currently `out`) first. I don't know if we want to rely on `make`s logic for determining when to rebuild a file.

Reply:
The build should be executed in a fresh copy of the source tree, so `out` should not exist yet. Pisek should scream when it exists (somebody has put it in the source tree). Pisek should also create `out` before running `make`, because directory dependencies in Makefiles are an unnecessary hassle.

Reply:
> The build should be executed in a fresh copy of the source tree

It's very possible that a users working tree will contain an `out` directory. How else are you meant to debug your program or Makefile? Or do you mean that the user should clean his worktree before running Pisek? Because that would render Pisek's caching pointless.

Reply:
I assumed that when users experiment with stand-alone builds, they also clean the trash (i.e., the destination directory, not the whole worktree).

Well, we might make it a bit more convenient by ignoring a pre-existing `out`, but it actually interferes with caching in a different way: the cache compares the whole contents of the source directory when deciding whether a rebuild is needed. Therefore if you leave anything in `out`, you invalidate this part of the cache. But I think it's a minor problem. (Still, I consider it good taste to clean the remnants of all experiments ;))

Reply:
This remains an issue.

Though we could put it off for later and fix it properly, say with a `.pisekignore` file, or some implicit strategy-defined exclude pattern.

Reply:
We have decided to put this off to a later time.